### ISSUE TITLE
Incorrect bbox conversion

### ISSUE THREAD

**Issue by 1-Olivier** [AcademySoftwareFoundation/openvdb#470]
`openvdb::math::Transform::indexToWorld(const CoordBBox&) ` is doing an incorrect conversion of the CoordBBox (which has inclusive integer coordinates) to a BBoxd. The resulting bbox is one unit too small in every axis.

**Comment by kmuseth** [comment]
This is not a bug, but rather a consequence of how the CoordBBox is designed. From the documentaiton of the math::CoordBBox class in math/Coord.h:

// @note The range of the integer coordinates, [min, max], is inclusive.
/// Thus, a bounding box with min = max is not empty but rather encloses
/// a single coordinate.

**Comment by 1-Olivier** [comment]
I understand how it happens. But how is it not a bug?`openvdb::math::Transform::indexToWorld(const CoordBBox&)` returns an incorrect bbox. As in the bbox is missing some volume.

Consider the case of a CoordBBox of [0, 0] which includes a single coordinate, as described in the documentation. If you call the above method, it gets converted to a BBoxd also which is also [0, 0] but in that case means empty. So the bbox which enclosed a single voxel is now empty.

The result is always one voxel too short in every axis, on the max side. I know because I wrote a renderer and had clipped volumes on 3 sides until I worked around it by doing my own conversion and calling the `indexToWorld(const BBoxd&) const` overload.
